htsget-search/src/htsget/mod.rs

```diff
@@ -135,7 +137,7 @@ impl Query {
   pub fn new(id: impl Into<String>) -> Self {
     Self {
       id: id.into(),
-      format: None,
+      format: Format::Bam,
```

Comment:
Why is BAM the default now?

Reply:
When it the default was None, it would perform BAM search anyway here:
https://github.com/umccr/htsget-rs/blob/ce002ef5ed4d97b2d721b89e73b34baa0d364625/htsget-search/src/htsget/from_storage.rs#L32
This basically isn't used because the format always gets set in htsget-http-core.

I left the behaviour the same because it didn't break any of the tests, even if the concept isn't quite correct. 

Interestingly the specific search struct (e.g. BamSearch, CramSearch, etc) more-or-less ignores this format and assumes that it is using the filetype of its name! It might make more sense to change this though, and just get it to return an `Htsget::UnsupportedFormat` error if passed the wrong type. This would be a pretty simple change.

Reply:
Oh, typical bioinformatics to rely on file extensions/filenames 🤦🏻 Yes please, change this so that it's a bit more robust & make sense, thanks for spotting it!